pkg/commands/compute/language_toolchain.go

```diff
@@ -179,6 +196,55 @@ func (bt BuildToolchain) Build() error {
 	return nil
 }
 
+// The encoding of a module starts with a preamble containing a 4-byte magic
+// number (the string '\0asm') and a version field.
+//
+// Reference:
+// https://webassembly.github.io/spec/core/binary/modules.html#binary-module
+func (bt BuildToolchain) validateWasm(path string) error {
+	// gosec flagged this:
+	// G304 (CWE-22): Potential file inclusion via variable
+	// Disabling as we trust the source of the path variable.
+	// #nosec
+	f, err := os.Open(path)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	defer f.Close()
+
+	// Parse the magic number
+	magic := make([]byte, wasmBytes)
+	_, err = f.Read(magic)
+	if err != nil {
+		return bt.handleError(err)
+	}
+	if len(magic) > 0 && magic[0] == 0x00 {
+		magic = magic[1:] // Remove the first byte which is a null character "\x00"
+	}
+	if string(magic) != "asm" {
+		return bt.handleError(err)
+	}
+	if bt.verbose {
+		text.Break(bt.out)
+		text.Description(bt.out, "Wasm module 'magic'", string(magic))
```

Comment:
And to add on to my comment above, the "magic" here is the 4 bytes including the nul, so it's important to print it out as such.

Reply:
I wasn't sure how best to display the value so opted for `fmt.Sprintf("%#v", magic)` which displays it as...

```
[]byte{0x0, 0x61, 0x73, 0x6d}
```

If I was to use `%v` instead of `%#v`, then it would have displayed as...

```
[0 97 115 109]
```

Which I didn't feel was as appropriate.

I also didn't want to hard-code the bytes into a string (e.g. `"0x00, 0x61, 0x73, 0x6d"`) as it means effectively having it defined twice (once in a hard-coded string and the other as the assignment to `expectedMagic`).

I also didn't want to have to `range` over the `magic` variable just to get it to display like `"0x00, 0x61, 0x73, 0x6d"` so opted for just `fmt.Sprintf("%#v", magic)`.